### PR TITLE
Adding Logging for non sdk based projects in VS and adding ErrorCode while logging errors in VS

### PR DESCRIPTION
**Pull request by mishra14**
Fixes: https://github.com/NuGet/Client.Engineering/issues/33

This PR addresses the following - 

1. Adds a `ProjectSettings` section in `PackageSpec` which is used to add a `HideWarningsAndErrors` property. This property is added as true for CPS/sdk based projects so that we do not throw errors on restore. We rely on the SDK target to throw errors post restore by reading them from the `AssetsFile`.

2. For cases where we do throw errors in VS (non-cps/sdk based projects), I have added an `ErrorCode` property to `ErrorListTableEntry` to display the error code in the `ErrorList` in VS. 

3. Improved Logging Extensions a bit.

### CODE REVIEW

**Comment by mishra14** [comment]
Waiting for PR approval, test run and Merge approval.

File: src/NuGet.Core/NuGet.Common/Logging/LoggingExtensions.cs
Comment (emgarten) [inline src/NuGet.Core/NuGet.Common/Logging/LoggingExtensions.cs:15]: 👍

File: src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs
Comment (emgarten) [inline src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs]: I like this idea, but since there is currently just 1 bool I think this should be a new class with only that property as a bool.

I also don't see being read/written which means that it will be lost during ``Clone()`` and that is standard for all dg spec operations.
https://github.com/NuGet/NuGet.Client/blob/d95e1fc5fae322cfab69e9d4c025ec49e3febbb1/src/NuGet.Core/NuGet.ProjectModel/PackageSpec.cs#L193-L204

When you add support for persisting this I think it will be more obvious that this needs to be a simple data type since you can't persist just any object to a json file.

**Review by emgarten — CHANGES_REQUESTED** [review]
Extension methods and VS logging looks great 👏 

The hide setting isn't persisted during clone or read/write which means it will get lost from the dg spec. Add tests for this.

**Comment by mishra14** [comment]
🔔 @emgarten

**Review by emgarten — APPROVED** [review]
HideWarningsAndErrors  -> hideWarningsAndErrors 

In the json, then :shipit:

**Pull request merged**
